### DISCUSSION TITLE
SQL syntax highlighting Inside Jupyter code cells

### DISCUSSION THREAD

**Discussion by ANP-Oxy** [quarto-dev/quarto-cli#9711] [Q&A] [answered]
### Description

I'm working on sql casestudies for my data analyst portfolio. I am using jupyter notebooks for this so I can also show output of the Sql queries in the case study. 

Is there a way to get SQL syntax highlighting inside the jupyter code blocks?

For example this is my code block. (it doesn't show syntanx highlighting like this in jupyter notebook)
```sql
%%sql 

SELECT 
    dannys_diner.sales.customer_id, 
    SUM(dannys_diner.menu.price) 
FROM 
    dannys_diner.sales
LEFT JOIN 
    dannys_diner.menu
ON dannys_diner.sales.product_id = dannys_diner.menu.product_id
GROUP BY dannys_diner.sales.customer_id;
```

I tried using quarto to see if I can get syntax highlighting
This is what my **_quarto.yml** looks like 

```yml
format:
  html:
    theme: 
      - Lux
    syntax-definations: 
      - sql
```

and the output in quarto preview looks like this
![image](https://github.com/quarto-dev/quarto-cli/assets/94713251/fb667d0e-c2c8-479d-ba81-888e879fa8d4)

I can just use .qmd document to get syntax highlighting too but I want to be able to show the query results to the reader that's why I'm using jupyter which let's me connect to databses using **ipython-sql** extension.

**Comment by mcanouil** [comment]
You need to create a SQL code block. Pandoc skylighting only works on code blocks.

https://github.com/quarto-dev/quarto-cli/assets/8896044/5c590ea0-9692-438b-835f-5ca8e32faded

**Reply by ANP-Oxy** [reply]
I did try this but when I preview the quarto document it doesn't show the syntax highlighting on the sql code block inside markdown either.

For now I'm just uploading the jupyter notebook to github and it shows syntax highligting on there so I'll just go with that option I guess. I'll close this discussion. Thank you for the reply.

**Reply by mcanouil** [reply]
You mean you want colour highlighting in the source code not in the output?
This highly depends on the editor and even with that said, for instance the VSCode extension does not map all other languages extension.


To summarise, the colour highlighting:
- in source code is provided by editors.
- in outputs code is provided by Pandoc Skylighting library.

In both cases, Quarto CLI is not in charge the same way as Python is not responsible for syntax highlighting in Jupyter.

**Reply by ANP-Oxy** [reply, marked as answer]
I wanted color highlighting in the output meaning the rendered output given by quarto. I figured out the solution. 
I had to add highlight-style argument in the configuration the default one wasn't working. 

```
format: 
    html:
        highlight-style: github
        theme: spacelab
        code-fold: true
        code-tools: true
        page-layout: full
```

now it looks like this in the output 

![image](https://github.com/quarto-dev/quarto-cli/assets/94713251/eaa3c82f-71de-413a-9bb8-90f7118c94d6)

**Reply by mcanouil** [reply]
FYI, that's documented in <https://quarto.org/docs/output-formats/html-code.html#highlighting>.
